sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
A few tests for complex numbers like those for the real will guard the new changes that you have made from future regression. Otherwise, I guess with any evidence to the contrary except mattpap's comments, I am +1.